zulipterminal/ui_tools/boxes.py

```diff
@@ -343,6 +378,7 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                         content=self.msg_write_box.edit_text
                     )
             else:
+                self.send_stop_typing_status()
```

Comment:
No, I missed taking out this call in the success block, I previously had an issue with it because emptying `edit_text` also sent a `start` status, but I've added the necessary check to avoid it.

Reply:
We could hoist this outside the conditional, like we do below, since the conditional is also in `send_stop_typing_status` now. Then this and the addition below wouldn't look conditional, and so applied in a similar way.

That also means if we changed the condition (stream, private, etc) then we'd just change it in that called function.